crates/router/src/connector/shift4.rs

```diff
@@ -495,13 +496,15 @@ impl api::IncomingWebhook for Shift4 {
     fn get_webhook_object_reference_id(
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
-    ) -> CustomResult<String, errors::ConnectorError> {
+    ) -> CustomResult<ObjectReferenceId, errors::ConnectorError> {
         let details: shift4::Shift4WebhookObjectId = request
             .body
             .parse_struct("Shift4WebhookObjectId")
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(details.data.id)
+        Ok(ObjectReferenceId::PaymentId(
+            api_models::payments::PaymentIdType::ConnectorTransactionId(details.data.id),
+        ))
```

Comment:
Need logic to detect type of webhook and return appropriate ObjectReferenceId as opposed to always returning PaymentId